### ISSUE TITLE
[Javascript] Implement new Concept Exercise: Function Expression and ES6 Arrow Function 

### ISSUE THREAD

**Issue by shubhsk88** [exercism/v3#2526]
This issue describes how to declare the function expression and declaration with the ES6 Arrow function. 

This issue describes how to implement the `function`  concept exercise for the Javascript track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This exercise aims to tell the student about the declaration of `function,` `arrow function.`

For example:
- `function hello ( ) { }`
- `let hello = function ( ) { }`
- `const hello = ( ) => { }`

## Learning objectives

- The student able to figure out the difference between function declaration and function expression.
- The student able to know how the arrow function can be implemented and where to use it.
-  Explaining the definition of high order function 

## Out of scope
- Closures
- Use of arrow function in the context of `this` keyword;
- Callbacks

##  Concepts

- `function` declartions

## Prerequisites
-  `functions-intro`
- `variables`

### After

- Understanding of execution context of the function 
- Higher-order function
- When to use function declarations or function expressions 

## Help

If you have any questions while implementing the exercise, please post the questions as comments on this issue.

**Comment by SleeplessByte** [comment]
Cool! 

I don't think higher order functions are in scope.

I think it will be hard to _test_ for use of one of the variants, BUT we can write an analyzer to do that instead.

Interesting would be:

- name of the function
- binding of this (as well as other context such as arguments)

Since this is a hard concept to teach clearly, I would recommend writing up ideas in this issue first before commuting a lot of time to it. 🤗
